### ISSUE TITLE
Spectral fitting within LightCurveEstimator

### ISSUE THREAD

**Issue by mcerruti** [gammapy/gammapy#3616]
I'd like to study spectral variability in AGN light-curves: I start from theoretical models and would like to extract, in addition to the integral light-curve, also the change in i.e. photon index or cut-off energy 

**Describe the solution you'd like**
Currently the LightCurveEstimator fits only the normalization of the model. I would like to have the option to also leave the index free  (or any other parameter of the input model)

**Describe alternatives you've considered**
I can of course loop over the models myself and compute indexes and integral fluxes per single simulated spectra, but it will much better if it is just an option of the light-curve core estimator 

**Additional context**

**Comment by adonath** [comment]
Thanks @mcerruti, indeed this is a very valid use case. Technically this is rather simple to implement as and extension to the `LightCurveEstimator`, but it requires a bit of work on the bookkeeping side. This means we have to extend the `FluxPoints` object to handle time varying reference models (which works by introducing a time axis to e.g. `FluxPoints.dnde_ref`) and add a data member to handle additional model parameters. Long term we probably also have to extend the serialization formats (https://gamma-astro-data-formats.readthedocs.io/en/latest/spectra/flux_points/index.html) to describe this. But as far as I can tell there is nothing that would block this from the Gammapy side.

**Comment by registerrier** [comment]
The estimate produced in such a context cannot be an extension of `FluxPoints`. It is a different object since it is mostly a collection of parameters (or even model) ordered with time. No energy binning is involved.

Isn't such a utility close to a possible `SkyModelEstimator` as discussed in #3702 . If an estimator is designed to evaluate a model component TS and the associated best fit parameters, it could directly be extended to perform the same analysis in time intervals, exactly as the `LightCurveEstimator` with the `FluxPointsEstimator`.

**Comment by maxnoe** [comment]
Sorry, I am not as familiar with the internals, but I thought with `reoptimize=True`, the `LightCurveEstimator` not only fits the amplitude but also the other parameters of the model. 

So obtaining index / break energy for a power law or powerlaw with cutoff already works with `LightCurveEstimator`, correct? 

Is the problem that this information is not stored when running the `LightCurveEstimator`?

**Comment by registerrier** [comment]
> Sorry, I am not as familiar with the internals, but I thought with `reoptimize=True`, the `LightCurveEstimator` not only fits the amplitude but also the other parameters of the model.
> 
> So obtaining index / break energy for a power law or powerlaw with cutoff already works with `LightCurveEstimator`, correct?
> 
> Is the problem that this information is not stored when running the `LightCurveEstimator`?

`reoptimize=True` will fit all free parameters that do not belong to the source (e.g. you don't want to refit source position or extension). The model of the source of interest is simply scaled by a `norm` factor. Getting the correct model is the responsibility of the user here. So, no the `LightCurveEstimator` cannot obtain index of cutoff. It has to be provided the correct spectral shape.

**Comment by maxnoe** [comment]
> reoptimize=True will fit all free parameters that do not belong to the source 

What are those parameters? At least for the normal case of a 1D spectral analysis light curve there are no parameters that are not of the source, correct? So for this, reoptimize=True or False does not make any difference?

**Comment by maxnoe** [comment]
Looking at this tutorial: https://docs.gammapy.org/0.19/tutorials/analysis/time/light_curve.html#Light-Curve-estimation:-by-observation

It's absolutely not clear to me, that 

> We can optionally ask for parameters reoptimization during fit

excludes optimizing parameters of the source model!

**Comment by maxnoe** [comment]
It even has a comment in the section of the code that defines frozen models that implies otherwise:

```
# Now we freeze these parameters that we don't want the light curve estimator to change
sky_model.parameters["index"].frozen = True
sky_model.parameters["lon_0"].frozen = True
sky_model.parameters["lat_0"].frozen = True
```

Why freeze these if they cannot be changed by `LightCurveEstimator` anyway?

**Comment by registerrier** [comment]
> It even has a comment in the section of the code that defines frozen model parameters that implies otherwise:
> 
> ```
> # Now we freeze these parameters that we don't want the light curve estimator to change
> sky_model.parameters["index"].frozen = True
> sky_model.parameters["lon_0"].frozen = True
> sky_model.parameters["lat_0"].frozen = True
> ```
> 
> Why freeze these if they cannot be changed by `LightCurveEstimator` anyway?

Indeed, this is useless and confusing. I suppose it dates from the early version of the `LightCurveEstimator` and was never changed.

**Comment by registerrier** [comment]
PR #4049 should alleviate the issues raised above with the `LightCurveEstimator` documentation

**Comment by registerrier** [comment]
A first part of this is solved in #5427 with the addition of tutorial showing how to perform this. Dedicated tools will be introduced later. Postponing to 2.0.

**Comment by AtreyeeS** [comment]
This API can be shared with #5732
